### ISSUE TITLE
Jackson ignores validation annotations for empty values with serialization include = NON_EMPTY

### ISSUE THREAD

**Issue by Davio** [FasterXML/jackson-databind#3329]
**Describe the bug**
Given a generated model with javax.validation annotations, such as `@NotNull` and an ObjectMapper with serialization inclusion set to `NON_EMPTY`. Jackson serializes to an invalid JSON string.

**Version information**
Which Jackson version(s) was this for? 2.13.0

**To Reproduce**
Create an ObjectMapper with serialization inclusion set to NON_EMPTY and create a JSON string:
```
class Scratch {

    public static void main(String[] args) throws Exception {
        var mapper = new ObjectMapper().setSerializationInclusion(NON_EMPTY).findAndRegisterModules();
        var test = new Test();
        var json = mapper.writeValueAsString(test);
        System.out.println(json); // prints '{}'
    }

    @Data
    private static class Test {

        @JsonProperty("values")
        @NotNull
        private List<String> values = new ArrayList<>();
    }
}
```

**Expected behavior**
Given that the property is annotated with `@NotNull` I would expect this to trump the serialization include and the property to always be present in the outputted JSON with an empty array: `{ "values": [] }`

**Additional context**
Note that this does not create any problems with Java applications serializing and deserializing these JSON objects as the receiver would just create the ArrayList itself if it does not find the property.

However, this can cause problems if the receiving party is a frontend JavaScript application. We use OpenAPI generator to generate the models for both Java and TypeScript. The definition looks like this:
```
Test:
      type: object
      required:
      - values
      properties:
        values:
          type: array
          items:
            type: string
```
The generated Java classes are similar to my example, any required property gets an `@NotNull` annotation. The TypeScript model gets a property without a `?` meaning the property is expected to always be present.

I understand that the validation annotations added by OpenAPI are not part of Jackson, they are Bean Validation API annotations, but it could be nice if Jackson could at least understand and take into account some of these annotations to ensure the generated JSON is valid.

**Comment by yawkat** [comment]
This seems fine to me? The field isn't present in the json, so it's not `null`.

**Comment by cowtowncoder** [comment]
Right, `@NotNull` is not a Jackson annotation and as such is not recognized. It cannot really be supported by default by databind since our goal is to keep mandatory dependencies to minimum.

But if someone wants to add support for this annotation, it's relatively easy to implement an `AnnotationIntrospector` that could use it to infer processing hints -- basically make it work like one of Jackson annotations, and change behavior.
In fact it is possible such a thing might exist, although I am not aware of such extension.

@Davio if you were interested in doing something like that, I would be happy to help. I don't remember off-hand which method(s) to implement in `AnnotationIntrospector`; existing `JacksonAnnotationIntrospector` is often a good source of information on what gets handled where.

FWTW, if such a module was written, eventually it could end up as a small module under

https://github.com/FasterXML/jackson-modules-base/

or such. It could handle a set of similar "non-null" annotations -- I think there are multiple packages in existence.
Module could also offer some configuration on exact semantics inferred.
